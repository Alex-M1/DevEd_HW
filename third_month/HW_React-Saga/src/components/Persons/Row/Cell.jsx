import React from 'react';
import PropTypes from 'prop-types';
import { StCell } from './styled';

function Cell({ id, type, children, onDoubleClick, updatePerson }) {
  const onBlur = (e) => {
    e.target.contentEditable = false;
    updatePerson({ id, type, value: e.target.textContent });
  };
  return <StCell onBlur={onBlur} onDoubleClick={onDoubleClick}>{children}</StCell>;
}

export default Cell;

Cell.propTypes = {
  id: PropTypes.string,
  type: PropTypes.string,
  children: PropTypes.string,
  updatePerson: PropTypes.func,
  onDoubleClick: PropTypes.func,
};
