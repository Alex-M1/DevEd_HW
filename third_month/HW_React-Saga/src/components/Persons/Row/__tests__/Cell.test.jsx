import { jest } from '@jest/globals';
import React from 'react';
import Cell from '../Cell';

describe('Cell', () => {
  const props = {
    id: 'sadas',
    type: 'sadas',
    children: 'sadas',
    onDoubleClick: jest.fn(),
    updatePerson: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = global.shallowSmart(<Cell {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StCell', () => {
    const component = global.mountSmart(<Cell {...props} />);
    expect(component.find('styled__StCell')).toHaveLength(1);
  });
  it('Should have call blur & contenteditable false', () => {
    const component = global.mountSmart(<Cell {...props} />);
    const item = component.find('styled__StCell');
    item.simulate('blur');
    expect(item.getDOMNode().contentEditable).toBe(false);
  });
  it('Should have call blur & call updatePerson', () => {
    const newValue = 'Mock';
    const component = global.mountSmart(<Cell {...props} />);
    const item = component.find('styled__StCell');
    item.simulate('blur', { target: { textContent: newValue } });
    expect(component.props().updatePerson).toHaveBeenCalledWith({
      id: props.id,
      type: props.type,
      value: newValue,
    });
  });
});
