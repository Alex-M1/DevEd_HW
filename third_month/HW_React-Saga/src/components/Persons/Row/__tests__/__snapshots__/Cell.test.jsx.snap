// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell Should match snapshot 1`] = `"<div class=\\"styled__StCell-sc-1gdbc6o-1 cSMJMA\\">sadas</div>"`;
